src/autograd/Functions.cpp

```diff
@@ -332,5 +340,141 @@ namespace af {
             };
             return Variable(result, {lhs, rhs}, grad_func);
         }
+
+        Variable unwrap(const Variable &input, int wx, int wy, int sx, int sy, int px, int py)
+        {
+            dim4 d = input.array().dims();
+            array res = unwrap(input.array(), wx, wy, sx, sy, px, py);
+            int params[] = {wx, wy, sx, sy, px, py, (int)d[0], (int)d[1]};
+            auto tmp = Variable(array(8, params), false);
```

Comment:
The `[]` part of the lambda need not be empty. You can specify what variables from the current scope are seen inside the lambda. http://en.cppreference.com/w/cpp/language/lambda#Lambda_capture

This does not change the signature of grad_func.